src/dataServices/dynamoDbDataService.ts

```diff
@@ -94,13 +98,26 @@ export class DynamoDbDataService implements Persistence, BulkDataAccess {
 
     async createResource(request: CreateResourceRequest) {
         const { resourceType, resource, id } = request;
+
         const vid = 1;
         let item = resource;
         item.resourceType = resourceType;
         item.meta = generateMeta(vid.toString());
 
         const params = DynamoDbParamBuilder.buildPutAvailableItemParam(item, id || uuidv4(), vid);
-        await this.dynamoDb.putItem(params).promise();
+        try {
+            await this.dynamoDb.putItem(params).promise();
+        } catch (e) {
+            if (e.code === 'ConditionalCheckFailedException') {
+                // It is highly unlikely that the autogenerated id will collide with a preexisting id, but adding that case
+                // to cover all bases
+                const errorMessage = id
+                    ? `Unable to create resource with id ${id}, because resource already exists`

```

Comment:
why? afaik we do NOT allow customers to choose their own id for resources. Are we changing that?

Not sure if that's by design though, all this code looks like we allow custom ids, but the router never passes an id to this function, so custom ids are not possible right now.

Reply:
Allowing the customer to set an id when creating a resource goes against the FHIE spec: https://www.hl7.org/fhir/http.html#create

Reply:
Since we don't allow customers to choose their own id, and FHIR specs doesn't allow that, I'll remove the `id` functionality from this method.

Reply:
Updated

Reply:
It is an Upsert that you can create with an ID: https://www.hl7.org/fhir/http.html#upsert

Reply:
But it looks like it is mapped to a PUT which would be routed in as an updateResource not create